### ISSUE TITLE
Upload file via HTTP

### ISSUE THREAD

**Issue by gabrielbiga** [ipkn/crow#46]
Hello all. Congrats for amazing project!

I need to know how i do one file upload with HTML form and save the file in my backend with crow.

Thanks.

**Comment by ipkn** [comment]
Currently crow cannot handle multipart data. We need to modify crow itself or to write a middleware for uploading a file.

**Comment by Geenz** [comment]
What would be nice is to at least have a stream for POSTs so larger multipart forms can be handled without having to deal with unnecessarily large strings for large file uploads.  At that point, something like MPFDParser could, to an extent, be a drop-in for a multipart form parser.

**Comment by Geenz** [comment]
@ipkn Could we at least get what I described in the interim?  Reason in being is, if someone really wanted to, they could just post a multi-gigabyte form to crow and crow's memory usage would reflect it, potentially resulting in denial of service attacks.  A stream would make it much more manageable in these situations.

**Comment by ipkn** [comment]
@Geenz I agree that the streaming is necessary, but it would be big changes to crow. How can we write a handler for the stream? How do middlewares handle such cases? What changes need to `Connection` class to support stream?

I was thinking about these questions so I forgot to answer. (I want to reuse the stream system for user handlers.) Simply, writing core module for uploading file (with the streaming support) is an easier way. Many other frameworks works this way. 

When the file uploading feature is implemented, it should limit the file length to avoid dos attack.

**Comment by talp101** [comment]
Hey, this feature is available?

**Comment by csrgxtu** [comment]
is crow support file upload?

**Comment by csrgxtu** [comment]
http://www.treefrogframework.org/ this cpp mvc framework support json return and file upload
